modules/node_modules/@ncigdc/dux/cart.js

```diff
@@ -92,6 +92,29 @@ function toggleFilesInCart(incomingFile) {
   };
 }
 
+function removeFilesFromCart(files) {
+  return (dispatch, getState) => {
+    const filesToRemove = _.isArray(files) ? files : [files];
+    const existingFiles = getState().cart.files;
+    const nextFiles = _.differenceBy(existingFiles, filesToRemove, 'file_id');
+    const filesRemoved = existingFiles.length - nextFiles.length;
+    dispatch(notify(getNotificationComponent('remove', `remove/${filesToRemove.file_name}`, {
```

Comment:
It works out for `toggleFilesInCart` because its being used for singular file so something like `filesToRemove.file_name` will always be unique. However for `addAllFilesInCart` or `removeFilesFromCart` where files could be in array the same logic doesn't work out. So either we would have to generate unique IDs or update notification to show it every time despite of what the ID is. This is where I think its being checked. Not totally sure though.

https://github.com/NCI-GDC/portal-ui/blob/relay/modules/node_modules/%40ncigdc/uikit/Notification.js#L101